csw-server/src/main/java/org/fao/geonet/component/csw/GetRecords.java

```diff
@@ -571,7 +571,7 @@ private String checkTypenames(Element query) throws MissingParameterValueEx, Inv
         }
         // missing typeNames element
         else {
-            return cswPrefix + ":Record";
+            throw new MissingParameterValueEx("typeNames","Attribute 'typeNames' is missing. Add typeNames=\"gmd:MD_Metadata\" to Query Element.");
```

Comment:
Reason is to comply with this test:
![gdi-de testsuite](https://cloud.githubusercontent.com/assets/2713718/7269673/3aa4ec3e-e8d6-11e4-8296-29dfa2b97cdd.PNG)

I will ammend the error message to state csw:Record of gmd:MD_Metadata and advice on the default value.